features/step_definitions/aruba_steps.rb

```diff
@@ -0,0 +1,65 @@
+if Aruba::VERSION < '1.0.0'
+  require 'aruba/cucumber/core'
+end
+require 'aruba/generators/script_file'
+
+When(/^I run `([^`]*)` in interactive mode$/) do |cmd|
+  cmd = sanitize_text(cmd)
+  @interactive = run_command(cmd)
+  @processed_inputs = []
+  @processed_outputs = []
+end
+
+Then(/^I should see a prompt "(.+?)"$/) do |prompt|
+  @interactive.stop
+  current_output = sanitize_text(@interactive.output)
+  expect(current_output).to include_output_string(prompt)
+end
+
+When(/^I type "([^"]*)" and continue$/) do |input|
+  @interactive.start
+  @processed_inputs << input
+  @processed_inputs.each do |step_input|
+    type(unescape_text(step_input))
+  end
+end
+
+Then /^the last (\d) lines of (output|stderr|stdout) should contain: "(.+?)"$/ do |line_count, channel, expected_output|
```

Comment:
As mentioned above, we only see the output of a command after it has finished, so we cannot return the output from each step of an interactive command. 
With this step we now have the following options:
Aruba: "the (output|stderr|stdout) should contain" --> all output
Aruba: "the (output|stderr|stdout) from "command" should contain" --> output of a specific command
Custom: "The last n lines of (output|stderr|stdout) should contain" --> last lines of all output

I'm not sure if this is what you imagined. If so, i could add another step for checking the last lines of a specific command

Reply:
Having to know how many lines of output to ignore is a low-level detail. While this could help to solve the "accidentally matched output of a previous command", a step to reset previous output would be most helpful. Something like `Given I ignore previous output`. Could this be added instead?

Reply:
If you only want to make sure that you do not accidentally match the output of another command, you could use the step `Then(/^(?:the )?(output|stderr|stdout) from "([^"]*)" should( not)? contain( exactly)? "([^"]*)"$/)`.  That only checks the output of the command you specify in the `from "([^"]*)"` part.

I thought, this step here could be useful if you don't want to check for whole text passages or sentences but only specific words or phrases, which may appear multiple times in an output, but you only want to see if they appear somewhere near the end, where they may be more relevant.

Reply:
But to also answer your question, yes it would be possible to reset the previous output. This would be achieved by resetting the `command_monitor` that holds all commands that have been run so far.

Reply:
Thanks for hinting that step, might be helpful. However, I still like an explicit `I ignore previous output`. Would you add that and remove all code related to "n lines" (including the helper at the bottom)?